hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/staking/RewardCalculator.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2022-2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl.staking;
+
+import static com.hedera.node.app.service.mono.ledger.properties.AccountProperty.BALANCE;
+import static com.hedera.node.app.service.token.Units.HBARS_TO_TINYBARS;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.state.token.StakingNodeInfo;
+import com.hedera.node.app.service.mono.ledger.accounts.staking.StakePeriodManager;
+import com.hedera.node.app.service.mono.ledger.properties.AccountProperty;
+import com.hedera.node.app.service.token.ReadableStakingInfoStore;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class RewardCalculator {
+    private final StakePeriodManager stakePeriodManager;
+    private final ReadableStakingInfoStore stakingInfoStore;
```

Comment:
This store should not be injected into constructor imo, since it is a singleton. We need to inject
into methods that are getting called

Reply:
Done